Modules/Sources/DesignSystem/Components/Modifiers/UILabel+DesignSystem.swift

```diff
@@ -0,0 +1,115 @@
+import UIKit
+
+// MARK: - UIKit.UIFont: TextStyle
+extension TextStyle {
+    var uiFont: UIFont {
+        switch self {
+        case .heading1:
+            return UIFont.DS.heading1
+
+        case .heading2:
+            return UIFont.DS.heading2
+
+        case .heading3:
+            return UIFont.DS.heading3
+
+        case .heading4:
+            return UIFont.DS.heading4
+
+        case .bodySmall(let weight):
+            switch weight {
+            case .regular:
+                return UIFont.DS.Body.small
+            case .emphasized:
+                return UIFont.DS.Body.Emphasized.small
+            }
+
+        case .bodyMedium(let weight):
+            switch weight {
+            case .regular:
+                return UIFont.DS.Body.medium
+            case .emphasized:
+                return UIFont.DS.Body.Emphasized.medium
+            }
+
+        case .bodyLarge(let weight):
+            switch weight {
+            case .regular:
+                return UIFont.DS.Body.large
+            case .emphasized:
+                return UIFont.DS.Body.Emphasized.large
+            }
+
+        case .footnote:
+            return UIFont.DS.footnote
+
+        case .caption:
+            return UIFont.DS.caption
+        }
+    }
+}
+
+// MARK: - SwiftUI.Text
+extension UILabel {
+    func style(_ style: TextStyle) -> Self {
+        self.font = style.uiFont
+        if style.case == .uppercase {
+            self.text = self.text?.uppercased()
+        }
+        return self
+    }
+}
+
+// MARK: - UIKit.UIFont
+fileprivate extension UIFont {
+    enum DS {
+        static let heading1 = DynamicFontHelper.fontForTextStyle(.largeTitle, fontWeight: .bold)
+        static let heading2 = DynamicFontHelper.fontForTextStyle(.title1, fontWeight: .bold)
+        static let heading3 = DynamicFontHelper.fontForTextStyle(.title2, fontWeight: .bold)
+        static let heading4 = DynamicFontHelper.fontForTextStyle(.title3, fontWeight: .semibold)
+
+        enum Body {
+            static let small = DynamicFontHelper.fontForTextStyle(.body, fontWeight: .regular)
+            static let medium = DynamicFontHelper.fontForTextStyle(.callout, fontWeight: .regular)
+            static let large = DynamicFontHelper.fontForTextStyle(.subheadline, fontWeight: .regular)
+
+            enum Emphasized {
+                static let small = DynamicFontHelper.fontForTextStyle(.body, fontWeight: .semibold)
+                static let medium = DynamicFontHelper.fontForTextStyle(.callout, fontWeight: .semibold)
+                static let large = DynamicFontHelper.fontForTextStyle(.subheadline, fontWeight: .semibold)
+            }
+        }
+
+        static let footnote = DynamicFontHelper.fontForTextStyle(.footnote, fontWeight: .regular)
+        static let caption = DynamicFontHelper.fontForTextStyle(.caption1, fontWeight: .regular)
+    }
+}
+
+private enum DynamicFontHelper {
```

Comment:
These exist in the app target and I duplicated them here and set them private. Does it make sense or would you rather another approach @Gio2018 ?

Reply:
Yes I definitely meant to ping you :D 

I didn't really want to refactor a ton of code and import Design System on all the call sites. Importing Design System for a helper that is heavily used in the codebase didn't feel right.